tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/aux/CherryPick.scala

```diff
@@ -590,9 +591,56 @@ class CherryPick(rewriter: SymbStateRewriter) {
     } else if (memberSets.forall(ms => state.arena.getHas(ms).isEmpty)) {
       // multiple sets that are statically empty, just pick the first one
       state.setRex(memberSets.head.toBuilder)
-    } else {
+    } else
       pickSetNonEmpty(cellType, state, oracle, memberSets, elseAssert, noSmt)
+  }
+
+  // Alternative method, with no interleaving. Unused at the moment, awaiting performance testing (#2415)
+  @unused
```

Comment:
This = method or this = @unused? The answer to the former is that I already had the method written (it was originally supposed to be this PR) and it would be a waste to just delete it and reinvent it later. The answer to the latter is https://github.com/informalsystems/apalache/actions/runs/4183474611/jobs/7247873624

Reply:
The former. Then could you keep it in a separate branch? In my view there's little use to temporarily merging an unused method if it might be removed again.

Also makes more sense to me to merge it as a separate logic unit, once we know that we want it in the codebase.

Reply:
Hmm, lets ask @konnov, to make the call; keep or remove?

Reply:
Good question. We could keep it on a separate branch as @thpani is suggesting. It should not be hard to introducing a branch and copy exactly this method over there.